base/compiler/methodtable.jl

```diff
@@ -0,0 +1,93 @@
+abstract type MethodTableView; end
+
+struct MethodLookupResult
+    # Really Vector{Core.MethodMatch}, but it's easier to represent this as
+    # and work with Vector{Any} on the C side.
+    matches::Vector{Any}
+    valid_worlds::WorldRange
+    ambig::Bool
+end
+length(result::MethodLookupResult) = length(result.matches)
+function iterate(result::MethodLookupResult, args...)
+    r = iterate(result.matches, args...)
+    r === nothing && return nothing
+    match, state = r
+    return (match::MethodMatch, state)
+end
+getindex(result::MethodLookupResult, idx::Int) = getindex(result.matches, idx)::MethodMatch
+
+"""
+    struct InternalMethodTable <: MethodTableView
+
+A singleton struct representing the state of the internal method table at a
```

Comment:
```suggestion
A struct representing the state of the internal method table at a
```

Reply:
Thanks. I originally had this a singleton struct with the world separately, but I think this makes more sense.